proposals/repo-to-tar/README.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Package Repository to Tar"
+authors: [ "Ashish Kumar <ashishndiitr@gmail.com>" ]
+status: "draft"
+approvers: [ ]
+---
+
+# <Package Repository to Tar>
+
+## Problem Statement
+
+Presently imgpkg does not support any way of creating and storing bundles in any format directly to local disk for sharing purposes, so as to remove the need of a registry in between. This proposal aims to add this functionality to imgpkg.
+
+Consider this workflow, where a user wants to create a tar file of a bundle image and then push it to a registry. <br>
+In order to obtain a tar file, one needs to first push the bundle image to the registry and then leverage the [command](https://carvel.dev/imgpkg/docs/v0.37.x/air-gapped-workflow/#option-2-with-intermediate-tarball) for copying as tar in the air gapped workflow.
+
+For example, In order to obtain a tar of bundle image of examples/basic-step-2, we need run the following commands in succession :<br>
+
+`imgpkg push -b index.docker.io/user1/simple-app-bundle:v1.0.0 -f examples/basic-step-2` <br>
+`imgpkg copy -b index.docker.io/user1/simple-app-bundle:v1.0.0 --to-tar /tmp/my-image.tar`
+
+This has 2 fold issues : <br>
+1. Dependent on using a registry to create a tar file from a bundle image.
+2. The tar file is created from the bundle image, which is not required if the user just wants to create a tar file having a layer as tar with configurations/metadata.
+
+## Terminology / Concepts
+1. Definitions of terms used in the proposal with respect to imgpkg can be found [here.](https://carvel.dev/imgpkg/docs/v0.37.x/)<br>
+2. Resources regarding docker save and docker load can be found [here.](https://docs.docker.com/engine/reference/commandline/save/)<br>
+3. A good article to understand OCI Artifacts can be found [here.](https://dlorenc.medium.com/oci-artifacts-explained-8f4a77945c13)
+
+## Proposal
+
+#### imgpkg
+imgpkg --tar-bundle-output command will be added to imgpkg. This command will create a tar file from the bundle image. The command will be used as follows : <br>
+`imgpkg --tar-bundle-output -b examples/basic-step-2 --to-tar /tmp/my-image.tar`<br>
```

Comment:
I believe we were inclined towards having `imgpkg save` as the command where as `--tar-bundle-output` points to the location.
And we would continue using `-f` to point towards folders.

So the command ends up looking like `imgpkg save -f examples/basic-step-2 --tar-bundle-output /tmp/my-image.tar`

We generally use commands to introduce functionalities and flags to pass arguments to the same.

Reply:
This ^ seems in line with the goals you listed below.

Reply:
Ah yes. Got overwritten in the update in the latest push ! Resolved